pprof/profile_builder.go

```diff
@@ -101,12 +124,12 @@ func (m *ProfileBuilder) AddExternalSampleWithLabels(locs []uint64, values []int
 	}
 	m.externalSampleID2SampleIndex[sampleID{locationsID: locationsID, labelsID: labelsID}] = uint32(len(m.Profile.Sample))
 	m.Profile.Sample = append(m.Profile.Sample, sample)
-	if len(labels) > 0 {
-		sample.Label = make([]*profilev1.Label, 0, len(labels))
-		for _, label := range labels {
+	if labelsCtx != nil && labelsSnapshot != nil {
+		sample.Label = make([]*profilev1.Label, 0, len(labelsCtx.Labels))
+		for k, v := range labelsCtx.Labels { //todo iterating over map is not deterministic, this can break tests and maybe even affect performance
```

Comment:
TODO: discuss if it is bad for stoarge

Reply:
We do not assume any particular order of labels at ingestion time and always sort them explicitly in the distributor https://github.com/grafana/pyroscope/blob/main/pkg/distributor/distributor.go#L270

Reply:
I think these are different labels. Distributor sorts series labels. These are sample labels.

Reply:
Yep, sample labels are sorted as well – at [normalisation](https://github.com/grafana/pyroscope/blob/main/pkg/distributor/distributor.go#L324) we drop duplicate samples based on their stacks and labels (ignoring numeric ones)

Reply:
ah, ok. Im probably overthinking it. Thanks